ocrd_eval.md

```diff
@@ -51,6 +43,7 @@ The Levenshtein distance between these texts is 4, because 4 edit operations are
 ##### Characters
 
 A text consists of a set of characters that have a certain meaning. A character is a glyph that represents a word, a letter in a word, or a symbol.
+Not included in the character definition are all forms of white spaces.
```

Comment:
Agreed. I can also add a sentence about the normalization being necessary.

Reply:
It's also not **realistic**. Most tools out there treat white space like any other codepoint. And most published evaluation results, too. 

It surely makes sense to provide more readily interpretable measures related to non-white-space or even letters (i.e. only alphanumerical codepoints). But that can only be an additional, optional service.

IMO this whole paragraph should focus on what **technically** is meant by characters. As in character as glyph and grapheme (think combining codepoints / grapheme clusters), or printable vs. non-printable.

The examples should highlight these problems, not trivial cases.

Reply:
> It's also not **realistic**. Most tools out there treat white space like any other codepoint. And most published evaluation results, too.

This passage has been adjusted after a discussion with @M3ssman . Maybe we can discuss this in the forum later.

> IMO this whole paragraph should focus on what technically is meant by characters. As in character as glyph and grapheme (think combining codepoints / grapheme clusters), or printable vs. non-printable.
>
> The examples should highlight these problems, not trivial cases.

I can add that, no problem.

Reply:
Summary of our discussion in the OCR-D forum: 

- in structured OCR data there is no white space included (pro omitting)
- a transcribed white space is a synthetic delimiter anyway, so we don't know how big the white space has been originally (pro omitting)
- omitting white space is not the default choice, though
- but to be interoperable it would be good to keep white spaces or offer both (with and without white spaces)
- how to identify errors like not detecting a white spaces (= words are not separated)?
- in order for the non-whitespace-approach to work you have to force OCR engines to work on words
- white space ambiguity is a main contributor to errors
- white spaces are the most common character in German language, thus they are a large source of errors. this is a good reason for keeping an eye on them

Since there a several good reasons for keeping the white spaces / consider them as character's I'll change that part according to our discussion result.

Reply:
Note that by introducing/adopting @M3ssman's **Letter Accuracy** (or Letter Error Rate), there would already be a metric which focusses on non-whitespace (and non-punctuation) characters.

Also (as I mentioned in the discussion), it could be prudent to mention the possibility to generate a confusion table (often in the form of the most common confusions/edits) along with the alignment. So this is not an additional "metric", but a diagnostic. In there one would quickly see the contribution of white space (word segmentation ambiguity) to the error.

Reply:
@bertsky Are there any implementations concerning this approach known / public available?

Reply:
Yesterday I came across empty regions - well, not *really* empty, the Workflow transcribed them as subsequent whitespaces. Source of confusion were delimiters and layout decorations. But now I don't want to argue pro omitting anymore, since this is rather a layout recognition issue. If the workflow didn't fail to properly divide characters from ornaments, it wouldn't fancy what characters this could be. And it might even be coherent if the system guesses in these cases a common space, considering it to be the most frequent character class after all.

Reply:
> Note that by introducing/adopting @M3ssman's **Letter Accuracy** (or Letter Error Rate), there would already be a metric which focusses on non-whitespace (and non-punctuation) characters.

@cneud, @kba and I talked about the Letter Accuracy yesterday. It's not part of our MVP but since it's a valuable addition we'll add it to the "not in use yet" section.

Reply:
Splendid. If you do add it, I suggest also mentioning the Confusion Table diagnostic there. (Which is also easy to implement BTW.)

Reply:
Resolving now since I updated this passage and added a placeholder section for LA (to be written yet).